config/src/config/mod.rs

```diff
@@ -177,52 +177,58 @@ impl NodeConfig {
 
     pub fn set_data_dir(&mut self, path: PathBuf) -> Result<()> {
         self.base = Arc::new(BaseConfig::new(path, self.base.role));
-        self.prepare()?;
+        self.prepare();
```

Comment:
btw not sure I understand the point of calling prepare every time, and if there is a point, what if someone forgets to call it?

Reply:
Its pretty fragile i agree, but if its an invariant that is maintained internally and not exposed then its ok

Reply:
Ahh, prepare ought not be public, good call... I plan to give this interface a bit of love sometime in the coming weeks once the dust settles here. I promise.

For now, let me make this API a bit more restrictive.

Reply:
I lied, they aren't in the same module scope any more. Perhaps let's table this as a TODO as part of the "giving some love in the coming weeks" initiative.